girder/events.py

```diff
@@ -223,18 +225,23 @@ def bind(eventName, handlerName, handler):
                     triggerer should be passed via the addResponse() method of
                     the Event.
     :type handler: function
+    :param overwrite: If this is overwriting an existing event binding intentionally,
+        set this to True to suppress the warning that would otherwise be logged.
+    :type overwrite: bool
     """
     if eventName in _deprecated:
-        girder.logger.warning('event "%s" is deprecated; %s' %
-                              (eventName, _deprecated[eventName]))
+        girder.logger.warning('event "%s" is deprecated; %s' % (eventName, _deprecated[eventName]))
 
     if eventName not in _mapping:
-        _mapping[eventName] = []
+        _mapping[eventName] = OrderedDict()
 
-    _mapping[eventName].append({
-        'name': handlerName,
-        'handler': handler
-    })
+    if handlerName in _mapping[eventName]:
+        # We delete from the OrderedDict to move the handler to the end of the ordering
+        del _mapping[eventName][handlerName]
+        if not overwrite:
+            girder.logger.warning('Overwriting event binding: %s -> %s' % (eventName, handlerName))
```

Comment:
I think it might be appropriate to raise an exception in this case because we are providing a mechanism to get around it.